gfdatasource/gfdatasource

```diff
@@ -103,19 +235,30 @@ def _split_creds(url):
     return url.geturl(), creds
 
 
+_cmds = {
+    'datasource': cmd_datasource,
+    'app': cmd_app,
+}
+
+
 def main():
     parser = make_parser()
     opts = parser.parse_args(sys.argv[1:])
     grafana_url, grafana_creds = _split_creds(opts.grafana_url)
     grafana_api = GrafanaAPI(base_url=grafana_url, credentials=grafana_creds)
-    datasource_url, datasource_creds = _split_creds(opts.data_source_url)
-    datasource = DataSource(
-        name=opts.name, type=opts.type, access=opts.access,
-        url=datasource_url, credentials=datasource_creds,
-    )
+
+    try:
+        cmd_func = _cmds[opts.cmd]
+    except KeyError:
+        print('Unknown command', opts.cmd)
+        sys.exit(1)
+
+    update_func = cmd_func(grafana_api, opts)
+
     while True:
-        grafana_api.update_datasource(datasource)
+        update_func()
```

Comment:
I didn't want to copy the while + sleep loop into both functions, but still avoid duplicate work.

It's probably overcomplicating things.